src/Interpreters/Context.cpp

```diff
@@ -1526,6 +1528,17 @@ BackgroundSchedulePool & Context::getDistributedSchedulePool() const
     return *shared->distributed_schedule_pool;
 }
 
+BackgroundSchedulePool & Context::getMessageBrokerSchedulePool() const
+{
+    auto lock = getLock();
```

Comment:
Is this for global Context only? Maybe an assertion would be helpful.

Reply:
Yes you are right, this is for global context only.

assertion can be added, but there are few other such helpers, that don't have it, and frankly speaking it is better to move such settings out from queries settings to the config directives, but let's not mix this